modules/common/src/main/java/com/ibm/cloud/cloudant/internal/ErrorTransformInterceptor.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * © Copyright IBM Corporation 2024. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.ibm.cloud.cloudant.internal;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.ibm.cloud.sdk.core.util.GsonSingleton;
+
+import okhttp3.Interceptor;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+public class ErrorTransformInterceptor implements Interceptor {
+
+    @Override
+    public Response intercept(Chain chain) throws IOException {
+        // Don't modify the request, but get the response
+        Response response = chain.proceed(chain.request());
+        if (!"HEAD".equals(chain.request().method()) // skip HEAD requests because they have no body
```

Comment:
This is redundant with `response.body() != null` below

Reply:
I would keep it if there's an event where the response of a non-HEAD request fails and the body doesn't exist.

Reply:
I mean it's other way around - we don't need HEAD check if we already have check for body not been `null` 😄

Reply:
I think @eiri means that we could skip the `HEAD` check and just use the body check, but if we want to do that we need to be sure on the any subtlty of empty string vs null

Reply:
Ok, I'm open to removing `HEAD` check.  If the body is not null and just empty, the GSON deserialize code should return a `null` response.  Here's the doc for `fromJson`:
```
Returns:
an object of type T from the string. Returns null if json is null or if json is empty.
```
If we don't want it to reach that code then we can add a `response.body().toString().isEmpty()` check.

Reply:
updated in [8411bd3](https://github.com/IBM/cloudant-java-sdk/pull/614/commits/8411bd3e6c47d464c766c8b15247cac4ee6bd51b)